Tools/CISupport/Shared/de-duper

```diff
@@ -114,21 +119,28 @@ def main(args):
             if os.stat(full_path).st_nlink > 1:
                 continue
             sha_value = sha1(full_path)
+            sha_path = os.path.join(args.storage, f'{sha_value}.png')
+            if sha_path == full_path:
+                continue
             if sha_value in existing_links:
                 if args.dry_run:
                     print(f'[dry-run] Linking {full_path} to {sha_value}.png')
                 else:
                     os.remove(full_path)
-                    os.link(os.path.join(args.storage, f'{sha_value}.png'), full_path)
+                    os.link(sha_path, full_path)
+                    existing_links[sha_value] += 1
+                    if existing_links[sha_value] > MAX_LINKS:
+                        os.remove(sha_path)
+                        del existing_links[sha_value]
                 de_duped += 1
             elif not sha_value:
                 print(f"Failed to compute SHA for '{full_path}'\n", file=sys.stderr)
             else:
                 if args.dry_run:
                     print(f'[dry-run] Storing and linking {full_path} to {sha_value}.png')
                 else:
-                    os.link(full_path, os.path.join(args.storage, f'{sha_value}.png'))
-                existing_links.add(sha_value)
+                    os.link(full_path, sha_path)
+                existing_links[sha_value] = 2
```

Comment:
I wonder if there's any world in which we get a different `inode` count… It would seem safer to read from the disk, again, especially rather than hardcoding this value?

Reply:
I was trying to avoid reading from disk again, but reading the rest of your comments, this feels like the correct design. The updated PR is removing the `existing_links` variable entirely in favor of reading from disk.